src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1005,10 +1010,26 @@ private void Resize(int newSize, bool forceNewHashCodes)
         /// Sets the capacity of a <see cref="HashSet{T}"/> object to the actual number of elements it contains,
         /// rounded up to a nearby, implementation-specific value.
         /// </summary>
-        public void TrimExcess()
+        public void TrimExcess() => SetCapacity(Count);
+
+        /// <summary>
+        /// Sets the capacity of a <see cref="HashSet{T}"/> object to the specified number of entries.
+        /// </summary>
+        /// <param name="capacity">The new capacity.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Passed capacity is lower than entries count.</exception>
+        public void TrimExcess(int capacity)
         {
-            int capacity = Count;
+            var currentCount = Count;
+            ArgumentOutOfRangeException.ThrowIfLessThan(capacity, currentCount);
 
+            if (capacity == currentCount)
```

Comment:
Similar question to https://github.com/dotnet/runtime/pull/85877#discussion_r1238581040. Even though it might be more expensive compared to other collections, my expectation is that if I call `set.TrimExcess(set.Count);` the capacity of the underlying hashtable should be shrunk to exactly match its count.

What does the equivalent method `Dictionary.TrimExcess()` do in that case?